src/lib.rs

```diff
@@ -119,10 +120,7 @@ where
                 }
             }
             Err(_) => {
-                hdr.append(
-                    "Server-Timing",
-                    HeaderValue::from_str(&header_value).unwrap(),
-                );
```

Comment:
Based on the `try_entry` docs, handling this `Err` does prevent a panic but doesn't increase the chance of being able to append the header.